src/csrf.ts

```diff
@@ -0,0 +1,85 @@
+/*
+ * @adonisjs/shield
+ *
+ * (c) ? (Please advice before merge. Thanks !)
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+*/
+
+/// <reference path="../adonis-typings/index.ts" />
+
+import Tokens from 'csrf'
+import { unpack } from '@poppinss/cookie'
+import { CsrfOptions } from '@ioc:Adonis/Addons/Shield'
+import { RequestContract } from '@ioc:Adonis/Core/Request'
+import { HttpException } from '@adonisjs/generic-exceptions'
+import { HttpContextContract } from '@ioc:Adonis/Core/HttpContext'
+
+import { noop } from './noop'
+
+const Csrf = new Tokens()
+
+// TODO: Switch this to fetch secret from session once questions have been answered.
+const session: { secret?: string } = {}
```

Comment:
You don't need the authentication package for this. `@adonisjs/session` is used to deal with sessions